### PR TITLE
[FIX] hr_gamification: Correct badge count on smart button

### PR DESCRIPTION
**Pull request by elhayyany**
The issue occurs when granting a badge to a user. The smart button always shows zero (0) badges granted, regardless of how many users have received the badge. This happens because the smart button displays the count of employees granted the badge, but `employee_id` is always `False` as a result counting how many `employee_id` granted this badge is always 0. `employee_id` it wasn't calculated. instead `user_id` was calculated from `employee_id`, but I think it was done by mistake.

I added the `_compute_granted_user_count()` function to count how many `users` have been granted the badge. This ensures the smart button now displays the correct number of users  and employees who received the badge. But I left the _compute_granted_user_count() function even while we don't needed it any more, just in case of someone who didn't update.

When clicking the smart button, now it lists all users that grant this budge, rather than listing only employees who grant it.

And the  `employee_id` is now computed, ensuring it is no longer `False` when the user is an employee.

**Steps to reproduce the issue:**
1: Install Employees
2: Install HR Gamification (not an app)
3: open Employees --> Configurations --> Challenges --> Budges
4: click on of those budges 
5: grant a budge to an employee
6: check the smart button it still shows that 0 employees were granted this budge
7: click on it and it wont list the employees who granted the budge

opw-4165187

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
Comparing to 16.0 where it works, the difference seems to be that in the wizard in 16.0 we hide the user_id field and only show the employee_id field if hr_gamification is installed:

https://github.com/odoo/odoo/blob/ac8c0b0e1dbeacfb7266518fb0f2c5e7ced59b48/addons/hr_gamification/wizard/gamification_badge_user_wizard_views.xml#L10-L15

so we will always have employee set. And the user_id will be computed from the employee.

In 17.0 we changed that and we still use the  user_id: 743e59d66687fd0fb73b1913f0f5520810e9d8b9

I think going with the change, we probably should do the change of field (granted_employees_count to granted_users_count in the view).

The `_compute_user_id` I think is used for people that did not update the view (since the change was done in stable) for the people that still have employee_id in the view, we want to still save the corresponding user_id for them. So it should be kept.

As for adding a `_compute_employee_id` so it's still filled, this might be ok. But I think what we want from ac8c0b0e1dbeacfb7266518fb0f2c5e7ced59b48 is to not have employee.

So in summary I think we should only do:

- no removal
- changing in the "hr_badge_form_view" the field granted_employees_count to granted_users_count
- maybe adding a _compute_employee_id (it would fix the button for future badges granting, for people that don't update hr_gamification)

File: addons/hr_gamification/models/gamification.py [hunk 1]
```diff
@@ -27,25 +27,33 @@ def action_open_badge(self):
...
     def _compute_granted_employees_count(self):
         for badge in self:
             badge.granted_employees_count = self.env['gamification.badge.user'].search_count([
                 ('badge_id', '=', badge.id),
-                ('employee_id', '!=', False)
             ])
 
-    def get_granted_employees(self):
-        employee_ids = self.mapped('owner_ids.employee_id').ids
+    def get_granted_users(self):
```
Comment (Bertrand2) [inline addons/hr_gamification/models/gamification.py:51]: we shouldn't change method signature in stable

**Review by Bertrand2 — CHANGES_REQUESTED** [review]
Thanks for the PR!

I'll discuss with our PO of what we should actually do with this :thinking: The link with employee on the badge seems totally useless IMO and should probably be dropped on a model definition basis. The smartbutton could also be moved in the base gamification module since it ends up being unrelated to employees, though those changes would have to be done in master.
As for this fix, I feel like if we want to change what's stated above, the fix should be limited to modifying `get_granted_employees` without changing the signature.
I'll get back to this PR as soon as I have a more decisive answer on that.

Last point is that I think it would be nice to add some kind of test to ensure that everything is working correctly :)

**Comment by elhayyany** [comment]
rd-hr will take this one
